internal/impl/devbox.go

```diff
@@ -1025,63 +1035,43 @@ func commandExists(command string) bool {
 	return err == nil
 }
 
-// leakedVars contains a list of variables that, if set in the host, will be copied
-// to the environment of devbox run/shell. If they're NOT set in the host, they will be set
-// to an empty value.
-// NOTE: we want to keep this list AS SMALL AS POSSIBLE. The longer this list, the less "pure"
-// (and therefore, reproducible) devbox becomes.
-// TODO: allow user to specify more vars to leak, in order to make development easier.
-var leakedVars = map[string]bool{
-	"HOME": true, // Without this, HOME is set to /homeless-shelter and most programs fail.
-
-	// Where to write temporary files. nix print-dev-env sets these to an unwriteable path,
-	// so we override that here with whatever the host has set.
-	"TMP":     true,
-	"TEMP":    true,
-	"TMPDIR":  true,
-	"TEMPDIR": true,
+// ignoreCurrentEnvVar contains environment variables that Devbox should remove
+// from the slice of [os.Environ] variables before sourcing them. These are
+// variables that are set automatically by a new shell.
+var ignoreCurrentEnvVar = map[string]bool{
+	// Devbox may change the working directory of the shell, so using the
+	// original PWD and OLDPWD would be wrong.
+	"PWD":    true,
+	"OLDPWD": true,
```

Comment:
what scenario leads to working directory changing?

I thought doing `devbox --config=/path/to/config shell` would still start you in the current directory but using the `devbox.json` of that config path. But maybe there's some other scenario I'm not considering.

Reply:
I think it's primarily plugin + init hooks that always run from the devbox.json directory.